### DISCUSSION TITLE
🎉 New Feature : Ideas and Feedback

### DISCUSSION THREAD

**Discussion by rabwill** [microsoft/copilot-camp#331] [Ideas]
We're exploring some cool updates! Share your thoughts on Feature X and let’s brainstorm together!

**Comment by Jerry-Winsten** [comment]
Can we bring in out of the box attachment feature within Copilot Studio bot creation. The attached files can be used by Gen AI answers and redaction of the document

**Comment by nssidhu** [comment]
I have a use case with the following points.
1. Ability for user to search Scanned structured PDF or images like Invoice from M365 Copilot UI ( Copilot.microsoft.com and also from withing Teams, Word Excel etc). As a search result user should be able to view the actual PDFs and Images
2. Another sue case is to search for image like find all the images which has trees in it, or find images which has accident, or find images which has license plate number ABC-123 etc.
3. Similarly ability to search Video as in point number 2, like seach a clip where person is wearing red shirt etc.
    As a search result user should be able to paly and view the video.
 4. Currently Copilot is limited to search very less file type. Implement Custom Copilot that can allow other type of files like Powet Point Excel etc to be searchable.
 
 All the above search should be performed over Documents in sharepoint,OneDrive or on Custom Location like Azure Storage.

**Comment by chenxizhang** [comment]
In the lab E2 (https://microsoft.github.io/copilot-camp/pages/extend-m365-copilot/02-build-the-api/), step 5, I'd suggest to not run the application at this moment, instead, you can guide user to just run the azure function locally. like run the command  `npm run dev` in the terminal.  @rabwill 

![image](https://github.com/user-attachments/assets/c3016030-e571-43cc-9dd4-9f17e93651cd)

**Comment by chenxizhang** [comment]
btw, Azure functions core tool is required for this lab, I've updated the doc and created a pull request for you. @rabwill

**Comment by hisandeepangara** [comment]
There seems to be an update for copilot agents built on Copilot Studio and which are deployed on Teams. Every response is coming up with thumbs up and thumbs down. Great feature, but is there a way to customise this or disable ?

**Comment by hanusatv** [comment]
I like really like the typespec approach of building agents.
I'm trying to make my own declarative agent and connect it to an MCP server, but I can't figure it out.

Could we get an example where a declarative agent has access to a MCP server with auth?
